### ISSUE TITLE
LSTM and GRU on cudnn with mask puts different output from CPU or Non-cudnn kernel

### ISSUE THREAD

**Issue by cosmoquester** [tensorflow/tensorflow#49241]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): colab
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): colab
- TensorFlow version (use command below): colab
- Python version: Python 3.7.10 (colab)
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: cuda_11.0_bu  (colab)
- GPU model and memory: colab

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`
v2.4.1-0-g85c8b2a817f 2.4.1

**Describe the current behavior**

When use LSTM or GRU with parameter `return_sequences=True` and use this layer on cudnn kernel,
The output values of timesteps on which input is masked is all zero.
And this behavior is different from CPU or GPU without cudnn, CPU  environment emit output of last unmasked timestep as masked timestep output.

**Describe the expected behavior**

I Think it should work same as it work on CPU or GPU without cudnn.

**[Contributing](https://www.tensorflow.org/community/contribute)** - Do you
want to contribute a PR? (yes/no): - Briefly describe your candidate solution

If it need python-level correction, I may contribute. But if it need cuda or cudnn something, I cannot contribute.
I Think the solution can be postprocessing for masked timestep simply.

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

Just run below code using colab on CPU and GPU.
The result is different.
```python
import tensorflow as tf

class TestModel(tf.keras.Model):
  def __init__(self):
    super().__init__()

    self.rnn = tf.keras.layers.LSTM(32, return_sequences=True)
  
  def call(self, x, mask=None):
    return self.rnn(x, mask=mask)[:, -1, :]

batch_size = 4
seq_len = 7

x = tf.random.normal([batch_size, seq_len, 32])
model = TestModel()
model(x, mask = tf.sequence_mask([seq_len - 1] * batch_size, maxlen=seq_len))
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by saikumarchalla** [comment]
Was able to reproduce the issue on GPU using TF version 2.4. Please find the gist [here](https://colab.research.google.com/gist/saikumarchalla/140720e835b6c5edc993261502547037/untitled.ipynb).

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
